src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineHelper.cs

```diff
@@ -349,13 +314,33 @@ public static void ExpandAncestors(ImmutableArray<DocumentSymbolUIItem> document
             }
         }
 
-        internal static void UnselectAll(IEnumerable<DocumentSymbolUIItem> documentSymbolItems)
+        internal static void UnselectAll(ImmutableArray<DocumentSymbolDataViewModel> documentSymbolItems)
         {
             foreach (var documentSymbolItem in documentSymbolItems)
             {
                 documentSymbolItem.IsSelected = false;
                 UnselectAll(documentSymbolItem.Children);
             }
```

Comment:
can all of these methods have threading asserts (it's fine if the assert says that it's safe to call on any thread).

Reply:
add threading assert please.